Add SV variables to BTV OfflineDQM

marco-link opened the pull request:
#### PR description:
- add secondary vertex variables to @miniAODDQM (removed with deprecated module in #34156)
- fix discriminator distributions to match with pre-#35985 distributions


Addition of SV variables presented in the [Btag WG Meeting](https://indico.cern.ch/event/1112381/#18-update-on-dqm-developments).

#### PR validation:
`runTheMatrix.py -l 12034.0`, used for validation plots to verify the fix of discriminator distributions.

`runTheMatrix.py -l limited -i all`

Commit:
add secondary vertex properties to OfflineDQM

Commit:
some label updates in BTV OfflineDQM

Commit:
add jet cuts to discriminators in BTV OfflineDQM

Commit:
fix handling of negative parton flavour

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36884/28147

- This PR adds an extra 32KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py  modified in PR(s): #36179

cmsbuild commented:
A new Pull Request was created by  @marco-link  for master.

It involves the following packages:

- DQMOffline/RecoB (**dqm**)
- PhysicsTools/PatAlgos (**reconstruction**)


@emanueleusai, @ahmad3213, @cmsbuild, @jfernan2, @clacaputo, @slava77, @jpata, @pmandrik, @pbo0, @rvenditti can you please review it and eventually sign? Thanks.
@AlexDeMoor, @rappoccio, @gouskos, @jdolen, @JyothsnaKomaragiri, @ahinzmann, @schoef, @emilbols, @jdamgov, @mbluj, @nhanvtran, @gkasieczka, @hatakeyamak, @gpetruc, @azotz, @mariadalfonso, @demuller, @rociovilar, @andrzejnovak, @seemasharmafnal, @mmarionncern this is something you requested to watch as well.
@perrotta, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

jfernan2 commented:
Hi @marco-link 
Before proceeding with this PR, is this PR related to the homework pending since last DQM Sequence review back in 2019:

https://indico.cern.ch/event/870363/contributions/3670597/attachments/1975300/3287456/BTV_DQM_Update_Final.pdf

It was agreed by then that, the old taggers SMT, SET, etc. could be removed from validation code for clean-up in code and memory.

Could this be performed now at some point? 
There is a thread with subject "Clean up of DQM BTV" which apparently you have not answered since almost 1 year ago
Thanks

marco-link commented:
Hi @jfernan2,
the old taggers have already been removed from the OfflineDQM/Validation by #34156.
Sorry if it was unclear that this also effects Validation.

This PR is a follow up to the cleanup in which I had to remove the SV variables because they were tied to an old tagger and were using a deprecated class.

jfernan2 commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22222/summary.html
**COMMIT**: 74ee2263158b9d6371847e158dff440469382035
**CMSSW**: CMSSW_12_3_X_2022-02-03-2300/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/36884/22222/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 5 differences found in the comparisons
 - DQMHistoTests: Total files compared: 46
 - DQMHistoTests: Total histograms compared: 3765022
 - DQMHistoTests: Total failures: 9998
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 3755001
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 50.884 KiB( 45 files compared)
 - DQMHistoSizes: changed ( 10024.0,... ): 1.696 KiB        Btag/SV
 - DQMHistoSizes: changed ( 312.0 ): 0.004 KiB        MessageLogger/Warnings
 - Checked 193 log files, 42 edm output root files, 46 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_3_X_2022-02-03-2300+c3ce97/48223/triggerResults): no differences found

jfernan2 commented:
@marco-link apart from the 13 new MEs added, there are many MEs which change their content with this PR. I don't have the expertise to judge if the differences are significant and in the right direction, could you please have a look?
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_3_X_2022-02-03-2300+c3ce97/48223/dqm-histo-comparison-summary.html
Thanks

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36884/28197

- This PR adds an extra 32KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py  modified in PR(s): #36890, #36179

cmsbuild commented:
Pull request #36884 was updated. @emanueleusai, @ahmad3213, @cmsbuild, @jfernan2, @clacaputo, @slava77, @jpata, @pmandrik, @pbo0, @rvenditti can you please check and sign again.

jpata commented:
are there any changes to MINIAOD size based on this?

marco-link commented:
@jpata Yes, there is a small increase in size caused by the addition of `pfSecondaryVertexTagInfos` to `slimmedJets`.

jpata commented:
can you please quantify the size increase with some meaningful sample size (few thousand PU events at least).

jfernan2 commented:
please abort
It looks like tests got stuck

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22323/summary.html
**COMMIT**: db0f057fe82757ac2e2a82909f42254aaba25792
**CMSSW**: CMSSW_12_3_X_2022-02-09-2300/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36884/22323/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



The workflows 1001.0, 1000.0, 136.88811, 136.874, 136.8311, 136.793, 136.7611, 136.731, 4.22 have different files in step1_dasquery.log than the ones found in the baseline. You may want to check and retrigger the tests if necessary. You can check it in the "files" directory in the results of the comparisons

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 46
 - DQMHistoTests: Total histograms compared: 3764420
 - DQMHistoTests: Total failures: 9483
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3754915
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 50.88 KiB( 45 files compared)
 - DQMHistoSizes: changed ( 10024.0,... ): 1.696 KiB        Btag/SV
 - Checked 193 log files, 42 edm output root files, 46 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_3_X_2022-02-09-2300+c3ce97/48303/triggerResults): no differences found

marco-link commented:
@jpata I checked the increase in file size with 10.000 events from `runTheMatrix.py -l 12034.0`:
It showed a 1% increase in file size (before: `334.155.602 bytes`, after: `337.624.186 bytes`) with the inclusion of `pfSecondaryVertexTagInfos`.

@jfernan2 comparison looks as expected now.

jfernan2 commented:
+1

jpata commented:
Thanks. 1% is significant for MINI and should be motivated by significant physics improvements. Does this need to exist for all and every MINI produced by CMS, or somehow more limited to DQM workflows?

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36884/28274

- This PR adds an extra 12KB to repository

cmsbuild commented:
Pull request #36884 was updated. @emanueleusai, @ahmad3213, @cmsbuild, @jfernan2, @clacaputo, @slava77, @jpata, @pmandrik, @pbo0, @rvenditti can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36884/28275

- This PR adds an extra 12KB to repository

cmsbuild commented:
Pull request #36884 was updated. @emanueleusai, @ahmad3213, @cmsbuild, @jfernan2, @pmandrik, @pbo0, @rvenditti can you please check and sign again.

marco-link commented:
> Thanks. 1% is significant for MINI and should be motivated by significant physics improvements. Does this need to exist for all and every MINI produced by CMS, or somehow more limited to DQM workflows?

@jpata Yes, it's only required for DQM and Validation workflows. I just figured out how to run it with the workflow.
I removed the commit with changes to the MiniAOD and cleaned up the rest.

jfernan2 commented:
please test

cmsbuild commented:
-1

**Failed Tests**: RelVals RelVals-INPUT 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/summary.html
**COMMIT**: c2a0f5a54cf267b5ac543d4cd7fa0772b7bb424f
**CMSSW**: CMSSW_12_3_X_2022-02-11-1400/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/36884/22364/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @jpata cms-sw/cmssw#36841

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/git-merge-result


## RelVals

- [136.8311](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/runTheMatrix-results/136.8311_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017)
```
----- Begin Fatal Exception 11-Feb-2022 21:16:56 CET-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 305064 lumi: 36 event: 55020723 stream: 0
   [1] Running path 'dqmoffline_step'
   [2] Prefetching for module MiniAODSVAnalyzer/'bTagSVDQM'
   [3] Calling method for module PATJetProducer/'patJetsSVInfo'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for a container with elements of type: reco::BaseTagInfo
Looking for module label: pfSecondaryVertexTagInfos
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```
- [136.88811](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/runTheMatrix-results/136.88811_RunJetHT2018D_reminiaodUL+RunJetHT2018D_reminiaodUL+REMINIAOD_data2018UL+HARVEST2018_REMINIAOD_data2018UL)
```
----- Begin Fatal Exception 11-Feb-2022 21:16:57 CET-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 320822 lumi: 17 event: 26886180 stream: 0
   [1] Running path 'dqmoffline_step'
   [2] Prefetching for module MiniAODSVAnalyzer/'bTagSVDQM'
   [3] Calling method for module PATJetProducer/'patJetsSVInfo'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for a container with elements of type: reco::BaseTagInfo
Looking for module label: pfSecondaryVertexTagInfos
Looking for productInstanceName: 

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```


## RelVals-INPUT

- [136.72411](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/runTheMatrixINPUT-results/136.72411_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM)```136.72411_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM/step2_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM.log```
- [136.72412](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/runTheMatrixINPUT-results/136.72412_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMININANO_data2016UL_HIPM+HARVESTDR2_REMININANO_data2016UL_HIPM)```136.72412_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMININANO_data2016UL_HIPM+HARVESTDR2_REMININANO_data2016UL_HIPM/step2_RunJetHT2016B_reminiaodUL+RunJetHT2016B_reminiaodUL+REMININANO_data2016UL_HIPM+HARVESTDR2_REMININANO_data2016UL_HIPM.log```
- [136.76111](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/runTheMatrixINPUT-results/136.76111_RunJetHT2016E_reminiaodUL+RunJetHT2016E_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM)```136.76111_RunJetHT2016E_reminiaodUL+RunJetHT2016E_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM/step2_RunJetHT2016E_reminiaodUL+RunJetHT2016E_reminiaodUL+REMINIAOD_data2016UL_HIPM+HARVESTDR2_REMINIAOD_data2016UL_HIPM.log```
<details>
<summary>Expand to see more relval errors ...</summary>

- [136.77211](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/runTheMatrixINPUT-results/136.77211_RunJetHT2016H_reminiaodUL+RunJetHT2016H_reminiaodUL+REMINIAOD_data2016UL+HARVESTDR2_REMINIAOD_data2016UL)
- [136.8311](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/runTheMatrixINPUT-results/136.8311_RunJetHT2017F_reminiaod+RunJetHT2017F_reminiaod+REMINIAOD_data2017+HARVEST2017_REMINIAOD_data2017)
- [136.83111](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/runTheMatrixINPUT-results/136.83111_RunJetHT2017F_reminiaodUL+RunJetHT2017F_reminiaodUL+REMINIAOD_data2017UL+HARVEST2017_REMINIAOD_data2017UL)
- [136.88811](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/runTheMatrixINPUT-results/136.88811_RunJetHT2018D_reminiaodUL+RunJetHT2018D_reminiaodUL+REMINIAOD_data2018UL+HARVEST2018_REMINIAOD_data2018UL)
- [140.5611](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/runTheMatrixINPUT-results/140.5611_RunHI2018AOD+RunHI2018AOD+REMINIAODHID18+HARVESTHI18MINIAOD)
- [1325.516](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/runTheMatrixINPUT-results/1325.516_TTbar_13_reminiaod2016UL_preVFP_INPUT+TTbar_13_reminiaod2016UL_preVFP_INPUT+REMINIAOD_mc2016UL_preVFP+HARVESTDR2_REMINIAOD_mc2016UL_preVFP)
- [1325.5161](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/runTheMatrixINPUT-results/1325.5161_TTbar_13_reminiaod2016UL_postVFP_INPUT+TTbar_13_reminiaod2016UL_postVFP_INPUT+REMINIAOD_mc2016UL_postVFP+HARVESTDR2_REMINIAOD_mc2016UL_postVFP)
- [1325.518](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/22364/runTheMatrixINPUT-results/1325.518_TTbar_13_reminiaod2018UL_INPUT+TTbar_13_reminiaod2018UL_INPUT+REMINIAOD_mc2018UL+HARVESTUP18_REMINIAOD_mc2018UL)
</details>

Commit:
disable secondary vertex OfflineDQM for some workflows

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-36884/28758

- This PR adds an extra 12KB to repository

cmsbuild commented:
Pull request #36884 was updated. @emanueleusai, @ahmad3213, @cmsbuild, @jfernan2, @pmandrik, @pbo0, @rvenditti can you please check and sign again.

soureek commented:
@jpata Can you please start testing the latest changes?? We would like to have it integrated ASAP.

jpata commented:
@soureek there are only DQM changes now, therefore I have not seen it in the reco list in the last weeks, and I have not commented on it.

@cms-sw/dqm-l2 can offer you a review and eventually sign.

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c3ce97/23178/summary.html
**COMMIT**: 188ec488d7b46f0bb47a979c6fb9dbbb37a47a19
**CMSSW**: CMSSW_12_4_X_2022-03-16-2300/slc7_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/36884/23178/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 49
 - DQMHistoTests: Total histograms compared: 3695377
 - DQMHistoTests: Total failures: 9518
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3685837
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 45.792 KiB( 48 files compared)
 - DQMHistoSizes: changed ( 10024.0,... ): 1.696 KiB        Btag/SV
 - Checked 204 log files, 45 edm output root files, 49 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_4_X_2022-03-16-2300+c3ce97/48985/triggerResults): no differences found

jfernan2 commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @perrotta, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

perrotta commented:
+1
